### PR TITLE
chore: cleanup API sender object

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
replace any type by unknown type
(and remove all eslint ignore/disable lines like @typescroipt-eslint/no-explicit-any)
also I had to fixes issues where we can't use async methods when sync methods are expected (before it was silently hidden due to the lack of signature/usage of any type)

make it consistent across all the packages and ensure it returns a disposable object for receiving events

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

consistency

### How to test this PR?

I've added a unit test on the implementation of Api Sender

But you could check that communication is working (UI is displaying correct objects ;-)

### CODE REVIEW

File: packages/main/src/plugin/contribution-manager.spec.ts [hunk 1]
```diff
@@ -42,14 +43,17 @@ const portNumber = 10000;
...
-const send = (channel: string, data?: any) => {
+const send = (channel: string, data?: unknown) => {
   eventEmitter.emit(channel, data);
 };
 
-const receive = (channel: string, func: any) => {
-  eventEmitter.on(channel, data => {
-    func(data);
-  });
+const receive = (channel: string, func: any): IDisposable => {
```
Comment (lstocchi) [inline packages/main/src/plugin/contribution-manager.spec.ts]: forgot to convert this any?

Comment (benoitf) [reply packages/main/src/plugin/contribution-manager.spec.ts]: ah yes it was in a test, will fix it

Comment (benoitf) [reply packages/main/src/plugin/contribution-manager.spec.ts]: fixed in a new commit

**Review by lstocchi — APPROVED** [review]
LGTM

**Pull request merged**